model/flow/timeout_certificate.go

```diff
@@ -21,22 +21,24 @@ type TimeoutCertificate struct {
 	SigData crypto.Signature
 }
 
-func (t *TimeoutCertificate) Body() interface{} {
+// ID returns the TimeoutCertificate's identifier
+func (t *TimeoutCertificate) ID() Identifier {
 	if t == nil {
-		return struct{}{}
+		return ZeroID
 	}
 
-	return struct {
+	body := struct {
 		View          uint64
 		NewestQCViews []uint64
-		NewestQC      QuorumCertificate
+		NewestQCID    Identifier
 		SignerIndices []byte
 		SigData       crypto.Signature
 	}{
 		View:          t.View,
 		NewestQCViews: t.NewestQCViews,
-		NewestQC:      *t.NewestQC,
+		NewestQCID:    t.NewestQC.ID(),
```

Comment:
⚠️  `QuorumCertificate` doesn't handle computing the ID of a `nil` QC, like we implemented for `TimeoutCertificate`, so we will panic if we compute the `ID` of a `TimeoutCertificate`. 

This is equivalent to assuming that we will always validate a `TimeoutCertificate` before computing its ID. I'm not convinced this is safe in general. For example, suppose we don't know the block for `tc.NewestQC` so we decide to cache the TC. Then we receive a new TC and want to see if it already exists in our cache, so we compare its ID to the cached TC's ID - that would cause a crash.

Reply:
agree, I think it would be fine to also define ID for a nil QC.

Reply:
Good catch, I've updated ID() to work with nil QC.